resources/assets/js/videos/components/thumbnailPreview.vue

```diff
@@ -101,22 +103,96 @@ export default {
+            prevImg.onload = () => {
+                this.preloadedSprites[prevSpriteUrl] = prevImg;
+            }
+            prevImg.onerror = () => {
+                if (prevSpriteUrl in this.triedUrls) {
+                    this.triedUrls[prevSpriteUrl]++;
+                }
+            }
+            prevImg.src = prevSpriteUrl;
+        },
+        preloadNextSprite() {
+            let nextIdx = this.spriteIdx + 1;
+            let nextSpriteUrl = this.getSpriteUrl(nextIdx);
+
+            if (this.spriteIdx === this.lastSpriteIdx
+                || nextIdx in this.preloadedSprites
+                || this.triedUrls[nextSpriteUrl] >= this.retryAttempts) {
+                return;
+            }
+            if (!this.triedUrls[nextSpriteUrl]) {
+                this.triedUrls[nextSpriteUrl] = 0
+            }
+            let nextImg = new Image();
+            nextImg.onload = () => {
+                this.preloadedSprites[nextSpriteUrl] = nextImg;
+            }
+            nextImg.onerror = () => {
+                if (nextSpriteUrl in this.triedUrls) {
+                    this.triedUrls[nextSpriteUrl]++;
+                }
+            }
+            nextImg.src = nextSpriteUrl;
+        },
+        removeOldSprites() {
+            let preloadedSprites = {}
+            for (let i = this.spriteIdx - 1; i < this.spriteIdx + 2; i++) {
+                if (i !== 0 || i !== this.lastSpriteIdx) {
+                    let url = this.getSpriteUrl(i);
+                    let img = this.preloadedSprites[url];
+                    if (img) {
+                        preloadedSprites[url] = img;
+                    }
+                }
+            }
+            this.preloadedSprites = preloadedSprites;
+        },
         updateSprite() {
-            this.spriteIdx = Math.floor(this.hoverTime / (this.thumbnailInterval * this.thumbnailsPerSprite));
-            let SpriteUrl = this.spritesFolderPath + "sprite_" + this.spriteIdx + ".webp";
+            let spriteUrl = this.getSpriteUrl(this.spriteIdx);
 
-            if (!this.triedUrls[SpriteUrl]) {
-                this.triedUrls[SpriteUrl] = 0
+            if (!this.triedUrls[spriteUrl]) {
+                this.triedUrls[spriteUrl] = 0
             }
-            if (this.triedUrls[SpriteUrl] < this.retryAttempts) {
-                this.sprite.src = SpriteUrl;
+            if (this.triedUrls[spriteUrl] < this.retryAttempts) {
+                if (spriteUrl in this.preloadedSprites && this.triedUrls[spriteUrl] === 0) {
+                    let onloadFunc = this.sprite.onload;
+                    let onerrFunc = this.sprite.onerror;
+                    this.sprite = this.preloadedSprites[spriteUrl];
+                    onloadFunc.call(this.sprite, new Event('load'));
+                    this.sprite.onload = onloadFunc;
+                    this.sprite.onerror = onerrFunc;
+                } else {
+                    this.sprite.src = spriteUrl;
+                }
```

Comment:
Shouldn't this rather be:
```
if spriteUrl in preloadedSprites
  -> use preloadedSprite
else if triedUrls[spriteUrl] < retryAttempts
  -> try to load sprite
else
  -> give up
```

Currently the preloaded sprite is only used if there were 0 retry attempts?

Reply:
With `this.triedUrls[spriteUrl] === 0` I tried to fix a bug. If you don't use this condition and hover really fast it can happen that the sprite index and sprite url don't belong to the same sprite any more.
I thought to only preload sprites which had no errors, because the `triedUrls` array is reset as soon as the cursor leaves the timeline. I assumed that it is not very likely for a user to keep the cursor on the timeline for a longer period without leaving it at least once.

Reply:
Regarding the structure, you are right. I forgot that the preload methods already check the `this.triedUrls[spriteUrl]` count.

Reply:
Because it was a while ago when I tried to debug this issue, I will take another look at it again and try to fix it.